projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.spec.ts

```diff
@@ -0,0 +1,148 @@
+            const defaultHeight = fix.debugElement.query(By.css('.igx-grid__tbody')).styles.height;
+            expect(defaultHeight).not.toBeNull();
+            expect(parseInt(defaultHeight, 10)).toBeGreaterThan(400);
+            expect(fix.componentInstance.isVerticalScrollbarVisible()).toBeTruthy();
+            expect(fix.componentInstance.treeGrid.rowList.length).toBeGreaterThanOrEqual(10);
+        }));
+
+        it(`should render all records exactly if height is 100% and parent container\'s height is unset and
+            there are fewer than 10 records in the data view`, fakeAsync(() => {
+                fix = TestBed.createComponent(IgxTreeGridWrappedInContComponent);
+                fix.componentInstance.treeGrid.height = '100%';
+                fix.componentInstance.data = fix.componentInstance.data.slice(0, 1);
+                tick();
+                fix.detectChanges();
+                const defaultHeight = fix.debugElement.query(By.css('.igx-grid__tbody')).styles.height;
+                expect(defaultHeight).not.toBeNull();
+                expect(parseInt(defaultHeight, 10)).toBeGreaterThan(200);
+                expect(fix.componentInstance.isVerticalScrollbarVisible()).toBeFalsy();
+                expect(fix.componentInstance.treeGrid.rowList.length).toEqual(6);
+        }));
+
+        it(`should render 11 records if height is 100% and parent container\'s height is unset and
+            display density is changed`, fakeAsync(() => {
+                fix = TestBed.createComponent(IgxTreeGridWrappedInContComponent);
+                fix.componentInstance.treeGrid.height = '100%';
+                fix.componentInstance.density = DisplayDensity.compact;
+                tick();
+                fix.detectChanges();
+                const defaultHeight = fix.debugElement.query(By.css('.igx-grid__tbody')).styles.height;
+                const defaultHeightNum = parseInt(defaultHeight, 10);
+                expect(defaultHeight).not.toBeNull();
+                expect(defaultHeightNum).toBeGreaterThan(300);
+                expect(defaultHeightNum).toBeLessThan(330);
+                expect(fix.componentInstance.isVerticalScrollbarVisible()).toBeTruthy();
+                expect(fix.componentInstance.treeGrid.rowList.length).toEqual(11);
+        }));
+    });
+
+});
+
+@Component({
+    template:
+        `<div [style.width.px]="outerWidth" [style.height.px]="outerHeight">
+            <igx-tree-grid #treeGrid [data]="data" [displayDensity]="density"
+                childDataKey="Employees" primaryKey="ID">
+                <igx-column [field]="'ID'" dataType="number"></igx-column>
+                <igx-column [field]="'Name'" dataType="string"></igx-column>
+                <igx-column [field]="'HireDate'" dataType="date"></igx-column>
+                <igx-column [field]="'Age'" dataType="number"></igx-column>
+        </igx-tree-grid>
+        </div>`
+})
+
+export class IgxTreeGridWrappedInContComponent {
```

Comment:
I would recommend moving the component definition to one of the test-util files containing other components.